bokeh/themes/theme.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import, print_function
+
+import yaml
+
+from ..properties import HasProps
+
+# whenever we cache that there's nothing themed for a class, we
+# use this same dict instance, so we don't have a zillion empty
+# dicts in our caches.
+_empty_dict = dict()
+
+class Theme(object):
```

Comment:
If `Theme` is eventually going to be a thing that can be sent to BokehJS, would it make sense to make it `Model`? Then there is also a natural place to validate the JSON for it too. Not sure.